weblate/addons/gettext.py

```diff
@@ -61,11 +62,26 @@ class GenerateMoAddon(GettextBaseAddon):
     description = _(
         'Automatically generates MO file for every changed PO file.'
     )
+    settings_form = GenerateMoForm
 
     def pre_commit(self, translation, author):
         exporter = MoExporter(translation=translation)
         exporter.add_units(translation.unit_set.all())
         output = translation.get_filename()[:-2] + 'mo'
+
+        if self.instance.configuration['path']:
+            mo_name = os.path.basename(output)
+            if mo_name:
+                mo_path = os.path.dirname(output)
+                if not mo_path:
+                    mo_path = ''
+
+                output = os.path.join(
+                    mo_path,
```

Comment:
Maybe it's better to use `self.instance.component.full_path` as root instead of trying to parse it from the filename?

Reply:
Well, that's a good point, I did consider that but I ultimately decided that since the current location is the directory of the filemask (locales/po/*.po, for example), using the component root may lead to confusion.  On the flip side, for those who were not used to the current method of working may expect that it is the component root being used as a base.

Should we include an option set for "Relative to component" / "Relative to PO file" ?  That way, we can use either method.

Reply:
I'd just document in the help text that it's relative path in the repository root. That should make it clear. But you will probably want to use template rendering then to allow using language code in the path...

Reply:
The thing is, if we just change it to relative to the repository root, it will move the location of existing .mo files.  For example, weblate generated a load for us in /locales/po/<lang>.mo to match the <lang>.po location.  If we just move that suddenly to repository root, it will make them /<lang>.mo whilst the po files are in /locales/po/<lang>.po.  That would surely break existing setups?